symengine/mp_class.h

```diff
@@ -123,9 +124,11 @@ inline void mp_set_str(integer_class &i, const std::string &a)
 
 inline std::string mp_get_hex_str(const integer_class &i)
 {
+    void (*freefunc)(void *, size_t);
+    mp_get_memory_functions(NULL, NULL, &freefunc);
```

Comment:
Is this needed? Would `mpz_clear`  not use any custom deallocator?

Reply:
I believe `mpz_clear` is only for `mpz_t` instances, because mpz seems to want to track how much memory was allocated/freed. 
https://gmplib.org/manual/Converting-Integers and https://gmplib.org/manual/Custom-Allocation.
It might work, but I'm not sure how correct it would be, since they go through different paths inside of gmp.
https://github.com/alisw/GMP/blob/2bbd52703e5af82509773264bfbd20ff8464804f/gmp-impl.h#L715-L716

Reply:
Sorry I misunderstood the situation. This is for the string allocation not for `mpz_t`